Add guidance for interruption panel contribution (#5262)
alphagov/govuk-design-system: src/patterns/interruption-pages/index.md

---
title: Interruption pages
description: Pause the user journey to give them important information
section: Patterns
theme: Pages
aliases: 
backlogIssueId: 27
layout: layout-pane.njk
---

> {% from "_example.njk" import example %}

Reviewer: I can see you've made an example but you haven't added it to the pattern page. You just need to call the example macro like so:

```nunjucks
{{ example({ group: "components", item: "panel", example: "default", html: true, nunjucks: true, open: false, size: "m", loading: "eager" }) }}
```

I suggest this goes below the lead paragraph but feel free to mess around with it.

description: Pause the user journey to give them important information
section: Patterns
theme: Pages
aliases: 
backlogIssueId: 27
layout: layout-pane.njk
---

{% from "_example.njk" import example %}

Use Interruption pages to pause the user’s journey and give them important information.

{{ example({ group: "patterns", item: "interruption-pages", example: "default", html: true, nunjucks: true, open: false, size: "l", loading: "eager" }) }}

## When to use this pattern

Think carefully before you choose to use an Interruption page, as doing this might get in the way of the user completing their task.

Use Interruption pages to warn the user before they:

- do something unusual that’s probably a mistake
- do something that cannot be undone (so you can ask the user “Are you sure?”)
- give an answer that conflicts with existing information

You can also use Interruption pages:

- instead of Confirmation pages to show more varied journey outcomes (that are different to a straightforward success, rejection or completion)
- to show the user important information ahead of a task

## When not to use this pattern

Do not use Interruption pages unless you're confident that both:

> - there’s evidence of a clear need to pause the user journey.

Author: ```suggestion
- there’s evidence of a clear need to pause the user journey
```

- it's the only way to give the user information and that they’d otherwise miss it

Read guidance in the Service Manual, particularly to [Map and understand a user's whole problem](https://www.gov.uk/service-manual/design/map-a-users-whole-problem), to find ways to improve the user journey and the organisational processes behind them.

## How it works

Interruption pages pause the user journey to show important information. Use Interruption pages sparingly, as they get less effective the more often users see them.

> All important information on the page is shown within an [Interruption panel](/components/panel/#interruption panel), to help ensure the user does not miss it.

Reviewer 2: ```suggestion
All important information on the page is shown within an [Interruption panel](/components/panel/#interruption-panel), to help ensure the user does not miss it.
```

The slug we're linking to cant have whitespace

> All important information on the page is shown within an [Interruption panel](/components/panel/#interruption panel), to help ensure the user does not miss it.

Author: This Markdown link is not showing on preview.

The panel includes a ‘continue’ button that the user must interact with to continue their journey. Information within the panel is usually shown as a heading with some description text, possibly with bullet points and numbered steps.

Keep information within the panel short. Do not place any other components or form elements inside the panel.

### Back links and cancel buttons

The Interruption page includes a back link at the top of a page, to allow users to go back to the previous page they were on.

If it’s helpful for users to jump back to another point in the journey, such as the start of a section, add a link and group it alongside the continue button. Write link text that describes the action it performs or the page it will take the user.

## Research on this pattern

> This pattern was originally contributed by the NHS Design System. We thank the team for working with us to publish the [Interruption panel variant of the Panel component](/components/panel/#interruption panel) and this pattern.

Reviewer 2: ```suggestion
This pattern was originally contributed by the NHS Design System. We thank the team for working with us to publish the [Interruption panel variant of the Panel component](/components/panel/#interruption-panel) and this pattern.
```

> This pattern was originally contributed by the NHS Design System. We thank the team for working with us to publish the [Interruption panel variant of the Panel component](/components/panel/#interruption panel) and this pattern.

Author: This Markdown link is not showing on preview.

---
title: Interruption pages
description: Pause the user journey to give them important information
section: Patterns
theme: Pages
aliases: 
backlogIssueId: 27
layout: layout-pane.njk
---

{% from "_example.njk" import example %}

> Use Interruption pages to pause the user’s journey to give them important information.

Author: ```suggestion
Use Interruption pages to pause the user’s journey and give them important information.
```

{{ example({ group: "patterns", item: "interruption-pages", example: "default", html: true, nunjucks: true, open: false, size: "l", loading: "eager" }) }}

## When to use this pattern

Think carefully before you choose to use an Interruption page, as doing this might get in the way of the user completing their task.

Use Interruption pages to warn the user before they:

- do something unusual that’s probably a mistake

> - do something that cannot be undone (asking the user “Are you sure?”)

Author: ```suggestion
- do something that cannot be undone (so you can ask the user “Are you sure?”)
```

Keep information within the panel short. Do not place any other components or form elements inside the panel.

### Back links and cancel buttons

The Interruption page includes a back link at the top of a page, to allow users to go back to the previous page they were on.

If it’s helpful for users to jump back to another point in the journey, such as the start of a section, add a link and group it alongside the continue button. Write link text that describes the action it performs or the page it will take the user. 

## Research on this pattern

This pattern was originally contributed by the NHS Design System. We thank the team for working with us to publish the [Interruption panel variant of the Panel component](/components/panel/) and this pattern.

Guidance in this pattern is based on:

- [Interruption page on NHS Design System](https://service-manual.nhs.uk/design-system/patterns/interruption-page)
- [Interruption card on Ministry of Justice (MOJ) Design System](https://design-patterns.service.justice.gov.uk/components/interruption-card/)

### Next steps

We’d like to get more input from service teams to help us further improve this pattern.

> See our [‘Interruption card’ discussion on GitHub](https://github.com/alphagov/govuk-design-system-backlog/issues/27) to see areas we're interested in.```

Author: ```suggestion
See our [‘Interruption card’ discussion on GitHub](https://github.com/alphagov/govuk-design-system-backlog/issues/27) to see some of the areas we're interested in learning more about.
```

## Research on this pattern

This pattern was originally contributed by the NHS Design System. We thank the team for working with us to publish the [Interruption panel variant of the Panel component](/components/panel/) and this pattern.

Guidance in this pattern is based on:

- [Interruption page on NHS Design System](https://service-manual.nhs.uk/design-system/patterns/interruption-page)
- [Interruption card on Ministry of Justice (MOJ) Design System](https://design-patterns.service.justice.gov.uk/components/interruption-card/)

### Other uses of Interruption pages

We’ve seen that many services in government use Interruption pages to show other types of important information that might otherwise get missed.

See our [‘Interruption card’ discussion on GitHub](https://github.com/alphagov/govuk-design-system-backlog/issues/27) to see these examples.

### Next steps

We’d like to get more feedback from service teams to help us improve this pattern.

> We'd particularly like to better understand the ways services use Interruption pages, so that we can provide better guidance on ‘When to use this pattern’ and ‘When not to use this pattern’ to ensure they're used effectively and consistently across government.

Author: Moves most of this information into a backlog post (draft), new issue TBC

```suggestion
This pattern was originally contributed by the NHS Design System. We thank the team for working with us to publish the [Interruption panel variant of the Panel component](/components/panel/) and this pattern.

Guidance in this pattern is based on:

- [Interruption page on NHS Design System](https://service-manual.nhs.uk/design-system/patterns/interruption-page)
- [Interruption card on Ministry of Justice (MOJ) Design System](https://design-patterns.service.justice.gov.uk/components/interruption-card/)

### Next steps

We’d like to get more input from service teams to help us further improve this pattern.

See our [‘Interruption card’ discussion on GitHub](https://github.com/alphagov/govuk-design-system-backlog/issues/27) to see areas we're interested in.```

---
title: Interruption pages
description: Pause the user journey to give them important information
section: Patterns
theme: Pages
aliases: 
backlogIssueId: 27
layout: layout-pane.njk
---

{% from "_example.njk" import example %}

Use Interruption pages to pause the user’s journey to give them important information.

{{ example({ group: "patterns", item: "interruption-pages", example: "default", html: true, nunjucks: true, open: false, size: "l", loading: "eager" }) }}

## When to use this pattern

Think carefully before you choose to use an Interruption page, as doing this might get in the way of the user completing their task.

Use Interruption pages to warn the user before they:

- do something unusual that’s probably a mistake
- do something that cannot be undone (asking the user “Are you sure?”)
- give an answer that conflicts with existing information

You can also use Interruption pages:

> - instead of Confirmation pages to show a more varied journey outcomes (that are different to a straightforward success, rejection or completion)

Reviewer 3: ```suggestion
- instead of Confirmation pages to show more varied journey outcomes (that are different to a straightforward success, rejection or completion)
```

- to show the user important information ahead of a task

## When not to use this pattern

> Do not use Interruption pages unless you're confident that both:

Reviewer 3: ```suggestion
Do not use Interruption pages unless you're confident that:
```

Author: Yeah, I know... saying "both" makes the lead-in sentence grammatically awkward, but there's a need to clarify that both points need to be met, and not either. I can't think of a better way to write it!

The [GOV.UK style guide for bullet lists](https://guidance.publishing.service.gov.uk/writing-to-gov-uk-standards/style-guides/a-to-z-style-guide/#bullet-points-and-steps:~:text=Bullets%20should%20normally%20form%20a%20complete%20sentence%20following%20from%20the%20lead%20text.%20But%20it%E2%80%99s%20sometimes%20necessary%20to%20add%20a%20short%20phrase%20to%20clarify%20whether%20all%20or%20some%20of%20the%20points%20apply.%20For%20example%2C%20%E2%80%98You%20can%20only%20register%20a%20pension%20scheme%20that%20is%20one%20of%20the%20following%3A%E2%80%99) allows for this bit of improper grammar:
> Bullets should normally form a complete sentence following from the lead text. But it’s sometimes necessary to add a short phrase to clarify whether all or some of the points apply. For example, ‘You can only register a pension scheme that is one of the following:’

> - there’s evidence of a clear need to interrupt the user journey.

Author: To [make consistent as per this comment](https://github.com/alphagov/govuk-design-system/pull/5262/changes#r3505092073).

```suggestion
- there’s evidence of a clear need to pause the user journey.
```

- it's the only way to give the user information and that they’d otherwise miss it

Read guidance in the Service Manual, particularly to [Map and understand a user's whole problem](https://www.gov.uk/service-manual/design/map-a-users-whole-problem), to find ways to improve the user journey and the organisational processes behind them.

## How it works

Interruption pages pause the user journey to show important information. Use Interruption pages sparingly, as they get less effective the more often users see them.

All important information on the page is shown within an Interruption panel, to help ensure the user does not miss it.

> The panel includes a ‘continue’ button that the user must interact with to continue their journey. Information within the panel is usually shown as heading with some description text, possibly with bullet points and numbered steps.

Reviewer 3: ```suggestion
The panel includes a ‘continue’ button that the user must interact with to continue their journey. Information within the panel is usually shown as a heading with some description text, possibly with bullet points and numbered steps.
```

Keep information within the panel short. Do not place any other components or form elements inside the panel.

### Back links and cancel buttons

The Interruption page includes a back link at the top of a page, to allow users to go back to the previous page they were on.

If it’s helpful for users to jump back to another point in the journey, such as the start of a section, add a link and group it alongside the continue button. Write link text that describes the action it performs or the page it will take the user. 

> Do not add links that duplicate the back link.

Reviewer 2: I wonder if it's still valid to duplicate because of WCAG consistent navigation which may argue that better to have consistent (maybe duplicate) navigation.

https://www.w3.org/WAI/WCAG21/Understanding/consistent-navigation.html

If that's the case maybe we shouldnt be saying to not duplicate the back link?

Author: Okay, had a chat and I'm going to suggest we remove this sentence and possibly put this as an open question under the 'Research' section.

There's a bit of a logic puzzle going on here between Consistent navigation and Link purpose.

```suggestion
```

## Research on this pattern

This pattern was originally contributed by the NHS Design System. We thank the team for working with us to publish the [Interruption panel variant of the Panel component](/components/panel/) and this pattern.

Guidance in this pattern is based on:

- [Interruption page on NHS Design System](https://service-manual.nhs.uk/design-system/patterns/interruption-page)
- [Interruption card on Ministry of Justice (MOJ) Design System](https://design-patterns.service.justice.gov.uk/components/interruption-card/)

### Other uses of Interruption pages

We’ve seen that many services in government use Interruption pages to show other types of important information that might otherwise get missed.

See our [‘Interruption card’ discussion on GitHub](https://github.com/alphagov/govuk-design-system-backlog/issues/27) to see these examples.

> 

Author: Adding a blurb about the cancel links, but this might not be useful at all.

```suggestion
### Other uses of Interruption pages

We’ve seen that many services in government use Interruption pages to show other types of important information that might otherwise get missed.

See our [‘Interruption card’ discussion on GitHub](https://github.com/alphagov/govuk-design-system-backlog/issues/27) to see these examples.

### Back links and cancel links that go to the same page

We'd like to better understand whether it's helpful to show users a cancel link that goes to the previous page, as this would duplicate the functionality of the back link. The cancel link's link text would likely also need to show the same link text as the back button to meet WCAG 2.2 success criteria on consistent navigtion and consistent identification.

```

Reviewer 2: I think this sort of thing better for the community backlog, my feeling is as per WCAG and design perspective:
1. having back link consistent is important
2. having the back link and 'cancel link' the same text is not (WCAG talks about the same components, they're not the same)

Author: Yeah, I agree we seem to be building up a bit of a backlog post. I'll work on drafting one with the aim of moving most of the stuff we've put in here in 'Research'
